### PR TITLE
Simplify snapcraft.io config

### PR DESCRIPTION
**Pull request by nottrobin**
This will require some other changes first.

Firstly, https://github.com/canonical-web-and-design/snapcraft.io/pull/2266 
needs to be merged and released so we don't any longer need
SENTRY_PUBLIC_DSN.

Secondly, we need to consolidate all relevant values in the `snapcraft-io`
secret.

QA
--

`./qa-deploy --production snapcraft.io`

### CODE REVIEW

**Comment by nottrobin** [comment]
Ready for review again

File: services/snapcraft-io.yaml [hunk 1]
```diff
@@ -50,47 +47,22 @@ spec:
                 configMapKeyRef:
                   name: environment
                   key: environment
-            - name: BSI_URL
-              valueFrom:
-                configMapKeyRef:
-                  name: snapcraft-io
-                  optional: true
-                  key: bsi_url
-            - name: BLOG_ENABLED
```
Comment (tbille) [inline services/snapcraft-io.yaml]: I think we need to keep this one

File: services/snapcraft-io.yaml [hunk 2]
```diff
@@ -50,47 +47,22 @@ spec:
                 configMapKeyRef:
                   name: environment
                   key: environment
-            - name: BSI_URL
```
Comment (tbille) [inline services/snapcraft-io.yaml]: I think we need to keep this one

File: services/snapcraft-io.yaml [hunk 3]
```diff
@@ -50,47 +47,22 @@ spec:
...
-              valueFrom:
-                secretKeyRef:
-                  name: snapcraft-io-config
-                  key: csrf_secret_key
-            - name: SENTRY_PUBLIC_DSN
-              valueFrom:
-                secretKeyRef:
-                  name: snapcraft-io-config
-                  key: sentry_public_dsn
-            - name: MARKETO_URL
```
Comment (tbille) [inline services/snapcraft-io.yaml]: we need to keep this

**Pull request merged**